### PR TITLE
meta-ledge-bsp: Bump u-boot and tf-a to latest stable master and fix various build errors for gcc > 8

### PR DESCRIPTION
**Pull request by apalos**
Patch 1: Cleanup u-boot for stm32mp1 and apply patches for proper DACR configuration
Pathes 2: bump TF-a to it's latest stable tag

Signed-off-by: Ilias Apalodimas <ilias.apalodimas@linaro.org>

### CODE REVIEW

**Comment by muvarov** [comment]
@apalos  Suggosh said me that efidebug syntax changed in the latest uboot. That means we need to update  BOOTCMD (auto boot uboot command.) We need to check that before the merge.

**Review by muvarov — CHANGES_REQUESTED** [review]
Patch "bsp: op-tee: Bump to 3.12" does not match description. And in general is not needed.

**Comment by apalos** [comment]
> Patch "bsp: op-tee: Bump to 3.12" does not match description. And in general is not needed.

Oops, the wrong patch crept in there somehow

File: meta-ledge-bsp/recipes-bsp/u-boot/u-boot-ledge/ledge_stm32mp157c_dk2_trusted_defconfig [hunk 1]
```diff
@@ -124,14 +123,24 @@ CONFIG_USB_GADGET_MANUFACTURER="STMicroelectronics"
...
+CONFIG_WDT_STM32MP=y
+CONFIG_ERRNO_STR=y
+# CONFIG_HEXDUMP is not set
 CONFIG_FDT_FIXUP_PARTITIONS=y
-CONFIG_EFI=y
-CONFIG_EFI_LOADER=y
-CONFIG_EFI_GET_TIME=y
 CONFIG_EFI_SET_TIME=y
-CONFIG_BOOTM_EFI=y
-CONFIG_CMD_BOOTEFI=y
```
Comment (muvarov) [inline meta-ledge-bsp/recipes-bsp/u-boot/u-boot-ledge/ledge_stm32mp157c_dk2_trusted_defconfig:133]: please double check that options are not needed with updated uboot.

Comment (apalos) [reply meta-ledge-bsp/recipes-bsp/u-boot/u-boot-ledge/ledge_stm32mp157c_dk2_trusted_defconfig:133]: Those are now default for u-boot, they are still included in our build

File: meta-ledge-bsp/recipes-bsp/u-boot/u-boot-ledge.bb [hunk 2]
```diff
@@ -9,16 +9,22 @@ PE = "1"
...
 SRC_URI = "git://git.denx.de/u-boot.git"
-SRCREV = "9f04a634ef331b4fc6b3e677f276156192a413c7"
+SRCREV = "c4fddedc48f336eabc4ce3f74940e6aa372de18c"
 
 SRC_URI += " \
-	file://0001-efi-change-usage-syntax.patch \
-	file://0002-STM32mp157c-DK2.patch \
-	file://0003-ti-am572x-enable-boot_distrocmd.patch \
-        file://ledge_stm32mp157c_dk2_trusted_defconfig \
+    file://0003-ti-am572x-enable-boot_distrocmd.patch \
```
Comment (muvarov) [inline meta-ledge-bsp/recipes-bsp/u-boot/u-boot-ledge.bb:18]: this patch is better to rename to 0000-ti-am...

**Pull request merged**